frappe/public/js/frappe/form/controls/link.js

```diff
@@ -487,49 +487,113 @@ frappe.ui.form.ControlLink = class ControlLink extends frappe.ui.form.ControlDat
+			// Handles 0/1, true/false values
 			if (docfield && docfield.fieldtype === "Check") {
-				filter[3] = filter[3] ? __("Yes") : __("No");
+				return (val == 1 || val === true ? __("Yes") : __("No")).bold();
 			}
 
-			if (filter[3] && Array.isArray(filter[3]) && filter[3].length > 5) {
-				filter[3] = filter[3].slice(0, 5);
-				filter[3].push("...");
+			// Array values -> truncate to first 5, append "..."
+			if (Array.isArray(val)) {
+				const filtered = val.filter((v) => v != null && v !== "");
+				const arr = filtered.slice(0, 5).map((v) => String(__(v)).bold());
+				if (filtered.length > 5) arr.push("...");
+				return arr.join(", ");
 			}
 
-			let value;
-			if (filter[3] && Array.isArray(filter[3])) {
-				value = filter[3].map((v) => String(__(v)).bold()).join(", ");
-			} else if (filter[3] == null || filter[3] === "") {
-				value = __("empty").bold();
-			} else {
-				value = String(__(filter[3])).bold();
+			// Null / empty
+			if (val == null || val === "") {
+				return String(__("empty")).bold();
 			}
 
-			return [__(label).bold(), __(filter[2]), value].join(" ");
+			// Default scalar
+			return String(__(val)).bold();
 		}
 
-		let filter_string = filter_array.map(get_filter_description).join(", ");
+		function get_operator_template(operator, fieldtype, value) {
+			// Special handling for Check fields
+			if (fieldtype === "Check") {
+				if (operator === "=" && value == 1) {
+					return __("{0} is enabled");
+				} else if (operator === "=" && value == 0) {
+					return __("{0} is disabled");
+				} else if (operator === "!=") {
+					return value == 1 ? __("{0} is disabled") : __("{0} is enabled");
+				}
+			}
+			// Return raw placeholder pattern for unknown operators (not translatable)
+			return operatorTemplates[operator] || "{0} {1} {2}";
+		}
+
+		function get_filter_description(filter) {
+			// expect [doctype, fieldname, operator, value]
+			const _doctype = filter[0];
+			const fieldname = filter[1];
+			const operator = filter[2];
+			let value = filter[3];
+
+			const docfield = frappe.meta.get_docfield(_doctype, fieldname);
+			const label = docfield ? docfield.label : frappe.model.unscrub(fieldname);
+			const fieldtype = docfield ? docfield.fieldtype : null;
+
+			// heuristics: detect a Check field if metadata absent but value looks boolean-ish
+			const looks_like_check =
+				(operator === "=" || operator === "!=") &&
+				(value == 1 || value == 0 || value === true || value === false);
+			const is_check_field = fieldtype === "Check" || looks_like_check;
+
+			// For Check fields with =/!= prefer human phrase without showing raw value
+			if (is_check_field && (operator === "=" || operator === "!=")) {
+				const template = get_operator_template(operator, "Check", value);
+				return __(template, [String(__(label)).bold()]);
+			}
+
+			const labelDisplay = String(__(label)).bold();
+			const valueDisplay = formatValueForDisplay(docfield, value);
```

Comment:
I personally would prefer _labels_ in italic; "string values" in quotes; numbers, dates and time not quoted:

> _Module_ equals "Core", _Allow Import_ is enabled, _Number_ is less than 5 and _Posting Date_ is between 1999-09-09 and 2022-02-02.

... as compared to:

> **Module** equals **Core**, **Allow Import** is enabled, **Number** is less than **5** and **Posting Date** is between **1999-09-09 and 2022-02-02**.